Sources/StreamChat/Database/DTOs/AttachmentDTO.swift

```diff
@@ -15,17 +15,22 @@ class AttachmentDTO: NSManagedObject {
     }
 
     /// An attachment type.
-    @NSManaged private var type: String
+    @NSManaged private var type: String?
     var attachmentType: AttachmentType {
-        get { .init(rawValue: type) }
+        get { AttachmentType(rawValue: type ?? AttachmentType.unknown.rawValue) }
         set { type = newValue.rawValue }
     }
 
     /// An attachment local state.
-    @NSManaged private var localStateRaw: String
+    @NSManaged private var localStateRaw: String?
     @NSManaged private var localProgress: Double
     var localState: LocalAttachmentState? {
-        get { LocalAttachmentState(rawValue: localStateRaw, progress: localProgress) }
+        get {
+            if let localStateRaw = localStateRaw {
+                return LocalAttachmentState(rawValue: localStateRaw, progress: localProgress)
+            }
+            return nil
```

Comment:
A nitpick with guard
```suggestion
            guard let localStateRaw = localStateRaw else {
                return nil
            }
            return nil LocalAttachmentState(rawValue: localStateRaw, progress: localProgress)
```

There's also shorter, one-liner version that imho goes against the readability & is quite confusing due to combination of ternary operator + force unwrap, so I don't recommend going that far :) 

```
localStateRaw != nil ? LocalAttachmentState(rawValue: localStateRaw!, progress: localProgress) : nil
```

Reply:
I have not touched this bit of code. But I would do it like this:
```
          return localStateRaw.flatMap { localStateRaw in
                return LocalAttachmentState(rawValue: localStateRaw, progress: localProgress)
            }
```